pkg/server/share_manager.go

```diff
@@ -220,8 +222,34 @@ func (m *ShareManager) runHealthCheck() {
 
 func (m *ShareManager) hasHealthyVolume() bool {
```

Comment:
Suggesting having two methods instead and change the code accordingly. The reason for the suggestion is you already change the behavior of `hasHealthyVolume`, so it does not just a status check function but also a self-healing mechanism inside. We should return the error back to the call, then log it by the caller instead of logging it in a status check method. 

```go
func (m *ShareManager) hasHealthyVolume() bool 
func (m *ShareManager) recoveryReadOnlyVolume() error 
```
then, change the check part in the runHealthCheck to below.

```go
			if m.hasHealthyVolume() {
                continue
            }
			
            if err := m.recoveryReadOnlyVolume() && err != nli {
                m.logger.WithError(err).Error("Volume health check failed, terminating")
				m.Shutdown()
				return
			}
```

Reply:
fixed.